cmd/samples/pageflow/main.go

```diff
@@ -0,0 +1,102 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"github.com/pborman/uuid"
+	"github.com/uber-common/cadence-samples/cmd/samples/common"
+	"go.uber.org/cadence/client"
+	"go.uber.org/cadence/worker"
+	"go.uber.org/cadence/workflow"
+	"time"
+)
+
+// This needs to be done as part of a bootstrap step when the process starts.
+// The workers are supposed to be long running.
+func startWorkers(h *common.SampleHelper) {
+	// Configure worker options.
+	workerOptions := worker.Options{
+		MetricsScope: h.WorkerMetricScope,
+		Logger:       h.Logger,
+	}
+	h.StartWorkers(h.Config.DomainName, ApplicationName, workerOptions)
+}
+
+func startWorkflow(h *common.SampleHelper) *workflow.Execution {
+	workflowOptions := client.StartWorkflowOptions{
+		ID:                              "pageflow_" + uuid.New(),
+		TaskList:                        ApplicationName,
+		ExecutionStartToCloseTimeout:    time.Minute,
+		DecisionTaskStartToCloseTimeout: 10 * time.Second,
+	}
+	return h.StartWorkflow(workflowOptions, pageWorkflow)
+}
+
+func queryState(h *common.SampleHelper, execution *workflow.Execution) *QueryResult {
+	var result QueryResult
+	err := h.ConsistentQueryWorkflow(&result, execution.ID, execution.RunID, QueryName, true)
+	if err != nil {
+		panic("failed to query workflow")
+	}
+
+	return &result
+}
+
+func assertState(expected, actual State) {
+	if expected != actual {
+		message := fmt.Sprintf("Workflow in wrong state. Expected %v Actual %v", expected, actual)
+		panic(message)
+	}
```

Comment:
Do we want to print something to indicate the workflow is making progress with expected values?

Reply:
Sure we can do that.